libraries/botbuilder-core/src/botAdapter.ts

```diff
@@ -158,6 +158,10 @@ export abstract class BotAdapter {
             revoke(): void;
         } = makeRevocable(context);
 
+        if (context && context.activity && context.activity.locale) {
+            context.locale = context.activity.locale;
+        }
```

Comment:
I'm confused about what this adds to the sdk?  Is there some code which has access to context.locale, but not context.activity.locale?

Reply:
no, all place which have access to context.locale should be able to access context.activity.locale. 

The drive of this change, is following @tomlm's suggestion at previous C# code review,  to make locale a property of "context".  The reason, from what i understand, to have a better model of the data. 

Creating a locale property in context directly **may** help decouple the locale from incoming activity.   Though today we set context.locale = context.activity.locale,  it might change overtime, so all the following process in the pipeline like (Dialog, LG, Expression) will be listening to context.locale, regardless whether it comes from activity or elsewhere, or user specified. Is that make sense to you? @EricDahlvang 

Add @tomlm for the original input,  and cc @carlosscastro

Reply:
@boydc2014 thank you for the explanation.